### PR TITLE
Feature/change weight calculation

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: app/ts/McdaApp/Workspace/CurrentScenarioContext/CurrentScenarioContext.tsx [hunk 1]
```diff
@@ -76,23 +77,36 @@ export function CurrentScenarioContextProviderComponent({
...
-        (result: AxiosResponse<IWeights>) => {
-          const updatedScenario = _.merge({}, scenario, {
-            state: {weights: result.data}
-          });
-          setCurrentScenario(updatedScenario);
-          updateScenarios(updatedScenario);
-        }
-      );
+      if (scenario.state.prefs[0]?.elicitationMethod === 'imprecise') {
+        const postCommand = getWeightsPataviProblem(
```
Comment (DanielReid) [inline app/ts/McdaApp/Workspace/CurrentScenarioContext/CurrentScenarioContext.tsx:81]: extract the if branches to fns?

File: app/ts/McdaApp/Workspace/ScenariosContext/PreferencesUtil.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import ICriterion from '@shared/interface/ICriterion';
```
Comment (DanielReid) [inline app/ts/McdaApp/Workspace/ScenariosContext/PreferencesUtil.ts:1]: i think the filename should start with lowercase 'p'

**Review by DanielReid — CHANGES_REQUESTED** [review]
ranking calculation to be changed

File: app/ts/McdaApp/Workspace/ScenariosContext/PreferencesUtil.ts (hunk 2 above)
Comment (chinook25) [reply app/ts/McdaApp/Workspace/ScenariosContext/PreferencesUtil.ts:1]: I agree!

**Pull request merged**